### PR TITLE
refactor(file-list): clean lib/file-list.js a bit

### PR DESCRIPTION
**Pull request by lusarz**

### CODE REVIEW

File: lib/file-list.js [hunk 1]
```diff
@@ -16,127 +13,54 @@ const helper = require('./helper')
...
-
-const byPath = (a, b) => {
+function byPath (a, b) {
   if (a.path > b.path) return 1
   if (a.path < b.path) return -1
 
   return 0
 }
 
-/**
```
Comment (devoto13) [inline lib/file-list.js]: This is and several comments below seem to have value as they explain implementation details. Why do you remove them?

Comment (lusarz) [reply lib/file-list.js]: I don't think these comments provide any value. Code should explain implementation details itself.

File: lib/file-list.js [hunk 2]
```diff
@@ -16,127 +13,56 @@ const helper = require('./helper')
...
-  // Check if we are currently refreshing
-  _isRefreshing () {
-    return this._refreshing.isPending()
+  _getFilesByPattern (pattern) {
+    return this.buckets.get(pattern) || []
   }
 
-  // Do the actual work of refreshing
   _refresh () {
-    const buckets = this.buckets
```
Comment (johnjbarton) [inline lib/file-list.js]: Are you confident that changing the delayed-update behavior is completely safe? 

In the old code `this.buckets` is not updated along several code paths. Seems like there could be some ways that this results in a change beyond a refactoring.  If so, I would prefer that the refactoring be separate so any behavior change is isolate to make debugging easier.

File: lib/file-list.js [hunk 3]
```diff
@@ -145,21 +71,20 @@ class FileList {
...
         return Promise.resolve()
       }
 
-      const mg = new Glob(pathLib.normalize(pattern), GLOB_OPTS)
+      const mg = new Glob(pathLib.normalize(pattern), { cwd: '/', follow: true, nodir: true, sync: true })
       const files = mg.found
-      buckets.set(pattern, new Set())
-
       if (_.isEmpty(files)) {
+        this.buckets.set(pattern, [])
```
Comment (johnjbarton) [inline lib/file-list.js]: Similarly here, it looks like the implied `else` branch now does not have value assigned for `pattern` and some paths eg rejects in the promise code below would now act differently.  (I'm not against pushing the change if the tests pass but mixing with refactoring is my issue)

File: lib/file-list.js (hunk 2 above)
Comment (lusarz) [reply lib/file-list.js]: In old code `this.buckets` was assigned to `buckets` variable at the beginning of `_refresh` method. In the whole `_refresh` method `this.buckets` and `buckets` point to the same object, so `buckets` variable seemed redundant to me - it's why I removed it.

`const buckets = this.buckets` is leftover from time when `arrow functions` was not used inside `_refresh` method.

**Pull request merged**